admin/src/cli.mts

```diff
@@ -0,0 +1,790 @@
+import BN from "bn.js";
+import { Command } from "commander";
+import Debug from "debug";
+import * as methods from "./methods.mts";
+import * as validators from "./validators.mts";
+import Client, * as clientHelpers from "./client.mts";
+import readSignerKeypair from "./utils/read-signer-keypair.mts";
+import resolveWalletPath from "./utils/resolve-wallet-path.mjs";
+import { populateSigners, prettifyJSON } from "./utils/index.mts";
+import { readJSON } from "./utils/read-file-content.mts";
+
+const VERSION = "0.1.0";
+
+const log = Debug("twamm-admin:cli");
+
+function handler(command: any, parser: any = prettifyJSON) {
+  return async (...args: any[]) => {
+    const res = await command(...args);
+    const out = parser ? await parser(res) : res;
+
+    console.log(out); // show the result via `console`
+  };
+}
+
+let cli = new Command()
+  .name("twamm-admin")
+  .description(
+    `Welcome to twamm admin. Use the "help" command to get more information.`
+  )
+  .requiredOption(
+    "-k, --keypair <path>",
+    "path to the payer's keypair; required"
+  )
+  .option("-u, --url <string>", "cluster address; supports monikers", "devnet")
+  .option(
+    "-d, --dry-run <bool>",
+    "allow to simulate the commands only",
+    (v) => v === "true"
+  )
+  .option("-f, --feature <test>", "specify feature flag for testing")
+  .version(VERSION);
+
+/**
+ * Read the global options and fill the `ANCHOR_WALLET`
+ * env variable with the path to look at.
+ */
+cli.hook("preSubcommand", (ctx, subCmd) => {
+  const { dryRun, keypair } = ctx.optsWithGlobals();
+
+  if (keypair) {
+    const ANCHOR_WALLET = resolveWalletPath(keypair);
+    Object.assign(process.env, { ANCHOR_WALLET });
+    log("`ANCHOR_WALLET` env was set to:", ANCHOR_WALLET);
+  } else {
+    log("`keypair` is absent. `ANCHOR_WALLET` should be set other way.");
+  }
+
+  if (dryRun) {
+    log(
+      "DryRun mode is enabled. Methods would be simulated instead of executing"
+    );
+  }
+});
+
+cli
+  .command("complete-orders")
+  .description("complete fulfilled orders")
+  .action(
+    handler(async (opts: unknown, ctx: Command) => {
+      const { dryRun, feature, keypair, url } = ctx.optsWithGlobals();
+      const client = Client(url);
+      const signer = await readSignerKeypair(keypair);
+
+      return methods.completeOrders(
+        client,
+        { options: { feature }, arguments: {} },
+        signer,
+        { dryRun }
+      );
+    })
+  );
+
+cli
+  .command("delete-test-pair")
+  .description("delete test pair")
+  .requiredOption("-tp, --token-pair <pubkey>", "Token pair address; required")
```

Comment:
for the single dash options, lets stick to a single letters, otherwise by looking at the command it is not clear if it is a single option or multiple merged (i.e. -t and -p merged into -tp)